### PR TITLE
Add hand raising using the UserData API

### PR DESCRIPTION
**Pull request by nienkedekker**
This PR adds hand raising to our demo 😄✋  it makes use of the new, still undocumented UserData API. This API is meant to allow participants to set and share custom participant state like if their hand is raised or they prefer hotdogs over hamburgers… anything! This data is tied to a single participant. Remotely, it's retrievable as part of the participant’s data. 

We're handling all hand raising action in a custom hook, `useHandRaising`. In `<Tray/>`, we set the controls to raise and lower hands. In `<Tile/>`, we show who has their hand up. 

~~This PR also updates daily-js and daily-react-hooks to their latest versions.~~ new PR for that: https://github.com/daily-demos/custom-video-daily-react-hooks/pull/15

### CODE REVIEW

**Review by jamsea — APPROVED** [review]
Looks good to me, looks like all the edge cases I can think of are covered 👍

File: src/hooks/useHandRaising.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    sort: useCallback((a, b) => {
+      const aDate = a.userData?.[HAND_RAISED_USER_DATA_KEY];
+      const bDate = b.userData?.[HAND_RAISED_USER_DATA_KEY];
+      if (aDate < bDate) return -1;
+      if (aDate > bDate) return 1;
+      return 0;
+    }, []),
+  });
+  if (idsWithRaisedHands.includes(id))
+    // arrays start with 0, but our queue numbers should start at 1
```
Comment (lazeratops) [inline src/hooks/useHandRaising.js:18]:
```suggestion
    // Arrays start with 0, but our queue numbers should start at 1
```

File: src/hooks/useHandRaising.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { useCallback } from 'react';
+import { useDaily, useParticipantIds } from '@daily-co/daily-react-hooks';
+export const HAND_RAISED_USER_DATA_KEY = 'hr'; // keep this as small as possible: the max payload size is 4K characters
```
Comment (lazeratops) [inline src/hooks/useHandRaising.js:3]:
```suggestion
export const HAND_RAISED_USER_DATA_KEY = 'hr'; // Keep this as small as possible: the max payload size is 4K characters
```

(ok I realize this is super nitpick, but we use capitalized comments in the rest of the file and it looks a bit neater :))

**Review by lazeratops — APPROVED** [review]
lgtm with just a couple of tiny formatting nits

**Comment by nienkedekker** [comment]
Closing this PR for now - I've received the feedback I needed. We don't want to make hand raising part of `main` until we finish the blog post and update the README :-)

**Pull request closed without merging**